src/portfft/dispatcher/subgroup_dispatcher.hpp

```diff
@@ -355,27 +367,32 @@ PORTFFT_INLINE void subgroup_impl(const T* input, T* output, const T* input_imag
         }
       }
       sycl::group_barrier(global_data.it.get_group());
-      if (SubgroupSize != factor_sg || output_batch_interleaved) {
+      if (!store_directly_from_private) {
         global_data.log_dump_local("computed data in local memory:", loc_view, n_reals_per_wi * factor_sg);
         // store back all loaded batches at once.
+        // data is batch interleaved in local
         if (!output_batch_interleaved) {
           global_data.log_message_global(__func__,
                                          "storing transposed data from local to global memory (SubgroupSize != "
                                          "FactorSG) with packed output layout");
           if (storage == complex_storage::INTERLEAVED_COMPLEX) {
-            detail::md_view local_md_view2{loc_view, std::array{2 * max_num_batches_local_mem, 1, 2}};
-            detail::md_view output_view{output, std::array{2, 1, 2 * fft_size}, i * n_reals_per_fft};
-            copy_group<level::WORKGROUP>(global_data, local_md_view2, output_view,
-                                         std::array{fft_size, 2, num_batches_in_local_mem});
+            const std::array<Idx, 3> local_strides{max_num_batches_local_mem * 2, 2, 1};
+            const std::array<IdxGlobal, 3> global_strides{output_stride * 2, output_distance * 2, 1};
+            const std::array<Idx, 3> copy_lengths{fft_size, num_batches_in_local_mem, 2};
```

Comment:
It looks like the final two values in these arrays have been switched. Why?

Reply:
Not completely intentional, it was easier to figure out the strides then start from the old ones. I do believe the final dimension of the strides is the "fast moving" one, so previously the work-items would read a `num_batches_in_local_mem` real values, then `num_batches_in_local_mem` imaginary values, which doesn't seem ideal to me. I think we want to have alternate work-items reading real and imaginary values, which should be what I've implemented if my understanding of `copy_group` is correct.

Reply:
That is true, but given our experience with global2local, it might actually be better for one workitem to load two consecutive values. Although we can not benchmark it right now, so I am fine with whichever option.

Reply:
I think the main point is that we're neither fixing nor introducing a bug! Thanks!

Reply:
> That is true, but given our experience with global2local, it might actually be better for one workitem to load two consecutive values. Although we can not benchmark it right now, so I am fine with whichever option.

I don't think we had that either way since `num_batches_in_local_mem` is usually `SubgroupSize`/2. I think that would lead to the first half of the workgroup loading the reals, and the second half loading imaginaries. You would need a "vector" version of `copy_group` like we have for `copy_wi`.